portable/GCC/ARM_CRx_MPU/portmacro_asm.h

```diff
@@ -0,0 +1,515 @@
+
+/** This following section is used to create the proper size for the ulContext array.
+ * This array is where all registers related to a task's context are saved.
+ * The size of this array will depend on if the system is using an integrated
+ * Floating Point Unit (FPU) or not. If we are using the FPU we must save the
+ * Floating Point Status and Control Register (FPSCR),
+ * and the Floating Point Registers (FPRs). The FPSCR holds the conditional bits
+ * used for floating point calculations. The FPRs hold the actual floating point bits.
+ * The remainder of a task's context consists of the General Purpose Registers (GPRs).
+ * General Purpose Registers are used to manipulate almost all variables.
+ * The Current Program Status and Control Register, which holds the operating mode
+ * and bits that correspond to any conditional checks, such as if statements.
+ * And the Critical Nesting Depth of the task.
+ *
+ *
+ * For more information about the FPU, FPSCR, and FPRs please reference ARM's website:
+ * https://developer.arm.com/documentation/den0042/a/Floating-Point
+ *
+ * Additional information related to the Cortex R4-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0363/e/fpu-programmer-s-model
+ *
+ * Additional information related to the Cortex R5-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0460/d/FPU-Programmers-Model
+ *
+ * Additional information related to the ArmV7-R CPSR
+ * https://developer.arm.com/documentation/ddi0406/cb/Application-Level-Architecture/Application-Level-Programmers--Model/The-Application-Program-Status-Register--APSR-?lang=en
+ *
+ * Additional information related to the GPRs:
+ * https://developer.arm.com/documentation/ddi0406/cb/System-Level-Architecture/The-System-Level-Programmers--Model/ARM-processor-modes-and-ARM-core-registers/ARM-core-registers?lang=en
+ *
+ */
+
+/** @brief The length in ulContext for the General Purpose Registers in bytes
+ * @note There are 13 GPRs, R0-R12 each is 32 bits, so 13 registers * 4 Bytes each
+ */
+#define portGPR_LENGTH                       ( 15U * 4U )
+
+/** @brief The length in ulContext for all the registers in a context
+ * @note There are the 13 GPRs, the Stack Pointer, and the Link Register
+ */
+#define portREGISTER_CONTEXT_LENGTH          ( ( 16 * 4U ) )
+
+/** If you KNOW that your system will not utilize the FPU in any capacity
+ * you can set portENABLE_FPU to 0, which will reduce the per-task RAM usage
+ * by ( 32 FPRs + 32 bit FPSCR ) * 4 bytes per register = 132, or 0x84, Bytes Per Task
+ * BE CAREFUL DISABLING THIS: Certain APIs will try and optimize themselves
+ * by using the FPRs. If the FPU context is not saved and this happens it could be
+ * exceedingly difficult to debug why a strcpy() or other similar function
+ * seems to randomly fail.
+ */
+#ifndef configENABLE_FPU
+    #define configENABLE_FPU 0
+#endif
+
+/** @brief Mark if the Floating Point Registers will be saved.
```

Comment:
Just as a suggestion, do you think it would add clarity if you had two version of this table, one which showed the packing when the FPU is used, and one without?
Or some other way to indicate how ulContext[] is sized/used based on the FPU configuration.

Reply:
I think it would!

The only reason I did it this way was so that if you hover over the variable in Visual Studio Code you then see that table

I've got no issue with adding the second one, I'll think of a way to reformat it to make that work

Reply:
Section has been re-formatted